ILL/IDF/d7_generateIDF.py

```diff
@@ -0,0 +1,120 @@
+import os
+path = os.path.abspath("")
+import sys
+sys.path.insert(0, path)
+from helper import MantidGeom
+
+from lxml import etree as le
+# using metre as unit
+instrumentName = 'D7'
+validFrom = "1900-01-31 23:59:59"
+validTo="2100-01-31 23:59:59"
+
+# Monochromator position
+source = -3.6
+
+# Fermi chopper
+zChopper = -0.48
+
+# 2 Monitors
+zMon1 = -1.3
+zMon2 = 1.5
+
+# distance between the sample and detectors:
+rSampleDetectors = 1.5177 # distance from the sample of even detectors
+minTwoTheta = 0
+maxTwoTheta = 136
+
+# definition of a detector bank
+numberBanks = 3 # banks 2, 3 and 4; bank 1 is a monitor
+numberDetectors = 44
+
+# definition of a pixel
+tubeRadius = 0.0125
+tubeHeight = 0.25
+
+# definitions of intersection positions for detectors in the same casette
+externalIntersectonDistance = 11.0
+intersectionAngle = 0.4189 # externalIntersectonDistance / (rSampleDetectors - tubeRadius) = 0.4189 degrees
+detectorOffset = 1.114 # degree
+
+
+comment = """ This is the instrument definition file of the D7 diffuse scattering spectrometer at the ILL.
+       Generated file, PLEASE DO NOT EDIT THIS FILE!
+       This file was automatically generated by mantidgeometry/ILL/IDF/d7_generateIDF.py
+       z axis defines the direction of the beam
+       y axis will be the axis used for rotation
+       coordinate system is right-handed
+       x axis rotation defined by phi 
+       y axis rotation defined by theta 
+       height y direction
+       The instrument operates in two modes
+       - Diffraction:
+          - wavelength 3.1 A, qmin = 0.4, qmax = 4.0
+          - wavelength 4.8 A, qmin = 0.3, qmax = 2.5
+          - wavelength 5.7 A, qmin = 0.2, qmax = 2.1
+       - Spectroscopy time-of-flight 
+          - using Fermi chopper, with the same possible wavelengths
+
+       Source-to-sample distances is 3.6 m 
+       Three detector banks with 44 position-insensitive He3 detectors each
+       Each detector located 1.5 m from the sample
+       covering 2 Theta range from about 10 to 155 degrees
+       Detector tubes have 25 mm in diameter and are 250 mm in height
+       For more information, please visit 
+       https://www.ill.eu/instruments-support/instruments-groups/instruments/d7/characteristics/
+       """
+d7 = MantidGeom(instrumentName, comment=comment, valid_from=validFrom)
+d7.addSnsDefaults(default_view='3D',axis_view_3d='z-')
+d7.addComment("SOURCE")
+d7.addComponentILL("Monochromator", 0., 0., source, "Source")
+d7.addComment("Sample position")
+d7.addComponentILL("sample_position", 0., 0., 0., "SamplePos")
+d7.addComment("Fermi chopper")
+d7.addComponentILL('fermi_chopper', 0.0, 0.0, zChopper, 'ChopperPos')
+d7.addComment("MONITORS")
+d7.addMonitors(names=["monitor1", "monitor2"], distance=[zMon1, zMon2])
+d7.addComment("MONITOR SHAPE")
+d7.addComment("FIXME: Do something real here.")
+d7.addDummyMonitor(0.01, 0.03)
+d7.addComment("MONITOR IDs")
+d7.addMonitorIds([repr(1000), repr(1001)])
```

Comment:
At the end let's make these 100000 and 100001, since D007 will have a 2D detector.

Reply:
OK